pkg/metrics/rate.go

```diff
@@ -35,11 +35,13 @@ func (r *Rate) flush(timestamp float64) ([]*Serie, error) {
 	}
 
 	value, ts := (r.sample-r.previousSample)/(r.timestamp-r.previousTimestamp), r.timestamp
+
+	origSample, origPreviousSample := r.sample, r.previousSample
 	r.previousSample, r.previousTimestamp = r.sample, r.timestamp
 	r.sample, r.timestamp = 0., 0.
 
 	if value < 0 {
-		return []*Serie{}, fmt.Errorf("Rate value is negative, discarding it (the underlying counter may have been reset)")
+		return []*Serie{}, fmt.Errorf("Rate value is negative (%f - %f = %f), discarding it (the underlying counter may have been reset)", origSample, origPreviousSample, value)
```

Comment:
How is this going to help us debug ? The only case this should happened is when the rate wrap (overflow) or the service we're monitoring is restarted.

Just thinking about having `(40 - 1254392 = -1254352)` in the log won't be very helpful, no ?

Reply:
In the case we are exploring now, the metrics are expected to be monotonically increasing, so having values which *do* decrease can offer some insight into what's happening, especially if it's a repeated behavior.

The numbers in your example wouldn't be helpful in isolation, but if we are seeing the same error for each check there might be some pattern which emerges.

We are also working on adding some extra debug to the integration code so we can see what gets sent into this Agent function.